lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1299,26 +1299,61 @@ getNetworkInformation (_block0, bp, st) nl = do
                 }
         }
   where
-    sp :: W.SlotParameters
-    sp = W.SlotParameters
-        (bp ^. #getEpochLength)
-        (bp ^. #getSlotLength)
-        (bp ^. #getGenesisBlockDate)
-        (bp ^. #getActiveSlotCoefficient)
+    sp = toSlotParameters bp
 
     -- Unsafe constructor for the next epoch. Chances to reach the last epoch
     -- are quite unlikely in this context :)
     unsafeEpochSucc :: HasCallStack => W.EpochNo -> W.EpochNo
     unsafeEpochSucc = fromMaybe bomb . W.epochSucc
       where bomb = error "reached final epoch of the Blockchain!?"
 
+toSlotParameters :: BlockchainParameters -> W.SlotParameters
+toSlotParameters bp = W.SlotParameters
+    (bp ^. #getEpochLength)
+    (bp ^. #getSlotLength)
+    (bp ^. #getGenesisBlockDate)
+    (bp ^. #getActiveSlotCoefficient)
+
 getNetworkParameters
-    :: forall t. ()
-    => (Block, BlockchainParameters, SyncTolerance)
-    -> NetworkLayer IO t Block
+    :: (Block, BlockchainParameters, SyncTolerance)
     -> ApiEpochNumber
     -> Handler ApiNetworkParameters
-getNetworkParameters _ _ _ = throwError err501
+getNetworkParameters (_block0, bp, _st) apiEpochNum = do
+    case apiEpochNum of
+        ApiEpochNumberInvalid txt ->
+            liftHandler $ throwE $ ErrGetNetworkParametersInvalidValue txt
+        ApiEpochNumber epochNum -> do
+            now <- liftIO getCurrentTime
+            let ntrkTip =
+                    fromMaybe slotMinBound (slotAt (toSlotParameters bp) now)
+            let currentEpochNum = ntrkTip ^. #epochNumber
+            when (currentEpochNum < epochNum) $ liftHandler $ throwE $
+                ErrGetNetworkParametersNoSuchEpochNo currentEpochNum epochNum
+            pure resp
+        ApiEpochNumberLatest ->
+            pure resp
+  where
+    (W.SlotLength slotLength) = bp ^. #getSlotLength
+    handlePercentageConv =
+        either
+        (error . ("from activeSlotCoefficient: " <>) . show)
+        Quantity . mkPercentage
+    resp :: ApiNetworkParameters
+    resp = ApiNetworkParameters
+        (ApiT $ bp ^. #getGenesisBlockHash)
+        (ApiT $ bp ^. #getGenesisBlockDate)
+        (Quantity slotLength)
+        (Quantity $ W.unEpochLength $ bp ^. #getEpochLength)
+        (bp ^. #getEpochStability)
+        ( handlePercentageConv
+            $ round @Double @Word
```

Comment:
I wonder if it is not a shame to round here.

https://github.com/input-output-hk/cardano-wallet/pull/1294#discussion_r370050003

I think I recall Matthias wondering in hindsight why the sync progress (in percentage) didn't have higher precision. ([here](https://input-output-rnd.slack.com/archives/GJY4XC3DM/p1578671245150200?thread_ts=1578668909.148800&cid=GJY4XC3DM))

Maybe we could think about changing the `Percentage`-type if it makes sense?

Reply:
Maybe best treated as separate U/S or issue

Reply:
yea, I was thinking about `Percentage` and its validity. The argument in favor of leaving it as it is is .... that it is like that in specification of the task in jira. I am kinda complying with it.

Reply:
I don't see how the U/S specifies it to be an integer though.

It's specified to be a `percentage`.

Whose quantity-field is a `number` https://github.com/input-output-hk/cardano-wallet/blob/master/specifications/api/swagger.yaml#L120-L134

`number` means _any number_ according to https://swagger.io/docs/specification/data-models/data-types/#numbers

Reply:
ok, I thought it refers to https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Data/Quantity.hs#L136 

I can change to Double - no problem:-)

Reply:
On the one hand, this field comes from double field of active slot coefficient, on the other when using Percentage (which is bounded enum of Word from 0 to 100 has its value that it is validated for free)...But if this field has differences important on 0.1 level then I am 100% persuaded we need to use double here.. Just do not know the value window and peculiarities of getActiveSlotCoefficient. I see it is 1% after rounding : 
```
[pawel@arch ~]$ curl http://localhost:8090/v2/network/parameters/latest --http1.1
{"slot_length":{"quantity":2,"unit":"second"},"epoch_stability":{"quantity":5,"unit":"block"},"genesis_block_hash":"f8c0622ea4b768421fea136a6e5a4e3b4c328fc5f16fad75817e40c8a2a56a56","blockchain_start_time":"2019-04-25T14:20:57Z","epoch_length":{"quantity":10,"unit":"slot"},"active_slot_coefficient":{"quantity":1,"unit":"percent"}}
```

Reply:
> ok, I thought it refers to

I would think that the reference in the U/S-swagger schema refers to the existing swagger schema rather than the Haskel type. It could be an overly mechanical reading though. I would understand if it was intended / interpreted as integer.

> I can change it to Double - no problem:-) Thought, that maybe we want to have additional validation (for free) stemming from the use of mkPercentage

Note (as said in DM also): I'm not saying that we should *definitely* do this. My comment was more arguing against the "it's specified as integer in the U/S" argument.

Reply:
after deep thought - will change to `activeSlotCoefficient :: !(Quantity "percent" Double)`